react/features/remote-video-menu/components/native/RemoteVideoMenu.js

```diff
@@ -0,0 +1,94 @@
+// @flow
+
+import React, { Component } from 'react';
+import { Text, View } from 'react-native';
+import { connect } from 'react-redux';
+
+import {
+    BottomSheet,
+    bottomSheetItemStylesCombined,
+    hideDialog
+} from '../../../base/dialog';
+import KickButton from './KickButton';
+import MuteButton from './MuteButton';
+import styles from './styles';
+
+type Props = {
+
+    /**
+     * The Redux dispatch function.
+     */
+    dispatch: Function,
+
+    /**
+     * The participant for which this menu opened for.
+     */
+    participant: Object
+}
+
+/**
+ * The exported React {@code Component}. We need it to execute
+ * {@link hideDialog}.
+ *
+ * XXX It does not break our coding style rule to not utilize globals for state,
+ * because it is merely another name for {@code export}'s {@code default}.
+ */
+let RemoteVideoMenu_; // eslint-disable-line prefer-const
+
+/**
+ * Class to implement a popup menu that opens upon long pressing a thumbnail.
+ */
+class RemoteVideoMenu extends Component<Props> {
+    /**
+     * Constructor of the component.
+     *
+     * @inheritdoc
+     */
+    constructor(props: Props) {
+        super(props);
+
+        this._onCancel = this._onCancel.bind(this);
+    }
+
+    /**
+     * Implements {@code Component#render}.
+     *
+     * @inheritdoc
+     */
+    render() {
+        const buttonProps = {
+            afterClick: this._onCancel,
+            showLabel: true,
+            participant: this.props.participant,
+            styles: bottomSheetItemStylesCombined
+        };
+
+        return (
+            <BottomSheet onCancel = { this._onCancel }>
+                <View style = { styles.participantNameContainer }>
+                    <Text style = { styles.participantNameLabel }>
+                        { this.props.participant.name }
```

Comment:
Is the participant guaranteed to have a name? Does it matter if the participant doesn't?

Reply:
No, you're right. I should have used the function that retrieves it with that special logic we use on the thumbnails.